### PR TITLE
Logger tests and fixes

### PR DESCRIPTION
**Pull request by ethanwharris**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes # (issue).

- Added several tests to loggers
- Fixed a couple of small bugs that were found in the process
- Improved coverage

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/mlflow.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None):
         for k, v in metrics.items():
             if isinstance(v, str):
                 logger.warning(
```
[the remark this answers is not in the document]
Comment (ethanwharris) [reply pytorch_lightning/loggers/mlflow.py]: Done :)

File: tests/loggers/test_base.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    assert logger.experiment[0] == mock1.experiment
+    assert logger.experiment[1] == mock2.experiment
+
+    logger.close()
+    mock1.close.assert_called_once()
+    mock2.close.assert_called_once()
+
+
+class CustomLogger(LightningLoggerBase):
```
[the remark this answers is not in the document]
Comment (ethanwharris) [reply tests/loggers/test_base.py]: I would but it's used in multiple tests, any other suggestions on a better place for it? Or could convert them into one test?

[the remark this answers is not in the document]
Comment (Borda) [reply tests/loggers/test_base.py]: Then is is fine =)

**Comment by Borda** [comment]
Great job!

**Pull request merged**